src/coreclr/src/gc/gc.cpp

```diff
@@ -16038,14 +15985,14 @@ int gc_heap::generation_to_condemn (int n_initial,
 
         if (check_max_gen_alloc)
         {
-            //figure out if large objects need to be collected.
-            if (get_new_allocation (max_generation+1) <= 0)
+            //figure out if non-soh objects need to be collected.
```

Comment:
replace non-soh with uoh?

Reply:
in theory this should loop through uoh as well but we can leave that for later as we'll likely have different tuning heuristics for poh

Reply:
I'll make this a loop. Here we just fetch and compare allocation budget. This part will probably be the same for poh. Budget might be different

---
In reply to: [351085027](https://github.com/dotnet/runtime/pull/235#discussion_r351085027) [](ancestors = 351085027)